src/main/java/com/minecolonies/entity/ai/citizen/builder/EntityAIStructureBuilder.java

```diff
@@ -785,13 +856,18 @@ private AIState findNextBlockNonSolid()
     @NotNull
     private AIState completeBuild()
     {
-        job.getSchematic().getEntities().forEach(this::spawnEntity);
+        if(job.getStructure() == null)
+        {
+            //fix for bad structures
+            job.complete();
+        }
+        job.getStructure().getEntities().forEach(this::spawnEntity);
 
-        String schematicName = job.getSchematic().getName();
+        String structureName = job.getStructure().getName();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'structureName' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)